azurerm/resource_arm_bot_connection.go

```diff
@@ -0,0 +1,262 @@
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name, botName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Bot Connection %q (Resource Group %q / Bot %q): %+v", name, resourceGroup, botName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_bot_connection", *existing.ID)
+		}
+	}
+
+	connection := botservice.ConnectionSetting{
+		Properties: &botservice.ConnectionSettingProperties{
+			ServiceProviderDisplayName: utils.String(d.Get("service_provider_name").(string)),
+			ClientID:                   utils.String(d.Get("client_id").(string)),
+			ClientSecret:               utils.String(d.Get("client_secret").(string)),
+			Scopes:                     utils.String(d.Get("scopes").(string)),
+			Parameters:                 expandAzureRMBotConnectionParameters(d.Get("parameters").(map[string]interface{})),
+		},
+		Kind:     botservice.KindBot,
+		Location: utils.String(d.Get("location").(string)),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if _, err := client.Create(ctx, resourceGroup, botName, name, connection); err != nil {
+		return fmt.Errorf("Error issuing create request for creating Bot Connection %q (Resource Group %q / Bot %q): %+v", name, resourceGroup, botName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, name)
+	if err != nil {
+		return fmt.Errorf("Error making get request for Bot Connection %q (Resource Group %q / Bot %q): %+v", name, resourceGroup, botName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Bot Connection %q (Resource Group %q / Bot %q): %+v", name, resourceGroup, botName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBotConnectionRead(d, meta)
+}
+
+func resourceArmBotConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ConnectionClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	botName := id.Path["botServices"]
+	name := id.Path["connections"]
+
+	resp, err := client.Get(ctx, id.ResourceGroup, botName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Bot Connection %q (Resource Group %q / Bot %q)", name, id.ResourceGroup, botName)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Bot Connection %q (Resource Group %q / Bot %q): %+v", name, id.ResourceGroup, botName, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("name", name)
+	d.Set("bot_name", botName)
+	d.Set("location", resp.Location)
+
+	if props := resp.Properties; props != nil {
+		d.Set("client_id", props.ClientID)
+		d.Set("scopes", props.Scopes)
+		if parameters := props.Parameters; parameters != nil {
```

Comment:
we can remove this if check and move it inside of the flatten function - since we should return an empty slice in that case so that `parameters` is always set into the state